laterpay/application/Helper/Voucher.php

```diff
@@ -60,18 +65,23 @@ public static function generate_voucher_code( $length = self::VOUCHER_CODE_LENGT
     }
 
     /**
-     * Save vouchers for current pass.
+     * Save vouchers for current pass / subscription.
      *
      * @param int   $pass_id
      * @param array $data
      * @param bool  $is_gift
+     * @param bool  $for_subscription
      *
      * @return void
      */
-    public static function save_pass_vouchers( $pass_id, $data, $is_gift = false ) {
-        $vouchers     = self::get_all_vouchers( $is_gift );
+    public static function save_vouchers( $pass_id, $data, $is_gift = false, $for_subscription = false ) {
```

Comment:
Same as `get_time_pass_subscription_vouchers` we should split this method into two.